smart-contracts/contracts/interfaces/hooks/ILockKeyPurchaseHook.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity 0.5.16;
+
+
+/**
+ * @notice Functions to be implemented by a keyPurchaseHook.
+ * @dev Lock hooks are configured by calling `setEventHooks` on the lock.
+ */
+interface ILockKeyPurchaseHook
+{
+  /**
+   * @notice Used to determine the purchase price before issueing a transaction.
+   * This allows the hook to offer a discount on purchases.
+   * @param from the msg.sender making the purchase
+   * @param recipient the account which will be granted a key
+   * @param referrer the account which referred this key sale
+   * @param data arbitrary data populated by the front-end which initiated the sale
+   * @return true if the user may purchase the key, false to cause purchase to revert
+   *         the minimum value/price required to purchase a key with these settings
+   * @dev the lock's address is the `msg.sender` when this function is called via
+   * the lock's `purchasePriceFor` function
+   */
+  function keyPurchasePrice(
+    address from,
+    address recipient,
+    address referrer,
+    bytes calldata data
+  ) external view
+    returns (bool purchaseSupported, uint minKeyPrice);
```

Comment:
Just a refresher, but we return a bool here rather than reverting to keep options open for how failures are handled right?

Reply:
Not sure if this is necessary... returning the bool is easier on JS integrations, but overkill for Solidity..

Reply:
Ok. I thought I remembered something( from the meeting about hooks )about wanting to allow a false value here to still be a valid response rather than just reverting. But I might be mixing this up with something else, that was a while ago and my mind has been elsewhere.

Reply:
I'm ok with a small optimization for js integrations here... it doesn't add much in the way of complexity or obfuscation.

Reply:
I think I'm going to switch back to using revert instead.. I think it's more clear. This is a reasonable use case to revert a view call.

Reply:
k switched, just waiting on ci